src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -2929,6 +2934,26 @@ private void OnRenderPackagesUpdated(RenderPackageCache packages)
                 RenderPackagesUpdated(this, packages);
             }
         }
+
+
+        public override void Dispose()
+        {
+            if (HasBeenDisposed) return;
+
+            base.Dispose();
```

Comment:
Minor point but `base.Dispose()` already handles the case where `HasBeenDisposed` is true.

Reply:
Yes but we want to check that on the nodemodel disposing as well as it calls the DisposePort() from inside here.